src/TTextEdit.cpp

```diff
@@ -1367,6 +1367,16 @@ void TTextEdit::mousePressEvent(QMouseEvent* event)
             popup->addAction(actionRestoreMainToolBar);
         }
 
+        // Add user actions
+        QMapIterator<QString, QStringList> it(mpHost->mConsoleActions);
+        while (it.hasNext()) {
+            it.next();
+            QStringList actionInfo = it.value();
+            QString actionName = actionInfo[1];
+            QAction * action = new QAction(actionInfo[1], this);
```

Comment:
No need to access the `QStringList` again - you have just extracted the value you wanted into another variable in the line above!
```suggestion
            QAction * action = new QAction(actionName, this);
```

Reply:
Also, be aware that providing the text in this manner may cause the Qt libraries to modify the text to add an accelerator which can silently change the stored text (which is another reason NOT to provide it from a non-`const` accessor!) See**[``(https://doc.qt.io/qt-5/qaction.html#QAction-1)** for some of the details...

Furthermore - you may wish to provide a further argument to the Lua `addMouseEvent` to permit a tool-tip to be provided - the default one (if tooltips on menus are enabled) for a menu is a repeat of the text supplied to the `QAction` constructor for the "display text" which is a bit pointless IMHO so provision for scripts/packages to supply tooltips (*if they are enabled - I cannot remember in this case*) is worth considering I think.

Reply:
Added the option to set tooltips. Otherwise they won't show up 👍